crates/stages/src/stages/execution.rs

```diff
@@ -138,6 +138,10 @@ impl<EF: ExecutorFactory> ExecutionStage<EF> {
         tx: &mut Transaction<'_, DB>,
         input: ExecInput,
     ) -> Result<ExecOutput, StageError> {
+        if input.target_reached() {
+            return Ok(ExecOutput::done(input.checkpoint()))
```

Comment:
as discussed, this missing check seems to have caused a bug where we're endlessly looping over this check due to the way `next_block` is calculated